prompting/tasks/base_task.py

```diff
@@ -59,6 +59,8 @@ class BaseTextTask(BaseTask):
     query_system_prompt: ClassVar[str | None] = None
     reference_system_prompt: ClassVar[str | None] = None
     augmentation_system_prompt: ClassVar[str | None] = None
+    dataset_entry: DatasetEntry | None = None
+    task_id: str = str(uuid4())
```

Comment:
Do we need to store task_id?

Reply:
the reason why I have it is that it makes things like logging etc. much more readable since once you introduce the async loops you can't know which task belongs to which scoring etc. without ids